packages/lib-sourcify/src/Compilation/auxdataUtils.ts

```diff
@@ -88,56 +88,75 @@ function getAuxdatasDiff(originalAuxdatas: string[], editedAuxdatas: string[]) {
+  //               REAL
+  // edited:   a2646970667358221220dceca8706b29e917dacf25fceef95acac8d90d765ac926663ce4096195952b6164736f6c63430007000033
+  //           └──────────────────┘└──────────────────────────────────────────────────────────────┘
+  //               DIFFSTART           DIFF
+  const auxdataDiffs = getAuxdatasDiff(originalAuxdatas, editedAuxdatas);
+
+  // 2. All bytecode character positions that differ after recompilation with slightly different sources
+  const bytecodeDiffPositions = getDiffPositions(
     originalBytecode,
     editedBytecode,
   );
-  const auxdataPositions: CompiledContractCborAuxdata = {};
 
-  let prevDiffPosition = -99;
-  for (const diffPosition of diffPositionsBytecodes) {
-    // Don't check consecutive diffs like 55, 56, 57... , only if there's a gap like 55, 57, 58, then 78, 79, 80...
-    if (prevDiffPosition + 1 === diffPosition) {
-      prevDiffPosition = diffPosition;
+  // 3.  For every bytecodeDiffPosition, see if it hosts an auxdataDiff
+  const result: CompiledContractCborAuxdata = {};
+  let prevPos = -Infinity;
+  for (const pos of bytecodeDiffPositions) {
+    // Skip positions that are immediately consecutive to the previous one,
+    // two auxdataDiffs can't be adjacent to each other
+    if (pos === prevPos + 1) {
+      prevPos = pos;
       continue;
     }
-    // New diff position
-    for (const auxdataDiffIndex in auxdataDiffObjects) {
-      const auxdataPositionsIndex = parseInt(auxdataDiffIndex) + 1;
-      if (
-        auxdataPositions[auxdataPositionsIndex] === undefined &&
-        bytecodeIncludesAuxdataDiffAt(
-          originalBytecode,
-          auxdataDiffObjects[auxdataDiffIndex],
-          diffPosition,
-        )
-      ) {
-        auxdataPositions[auxdataPositionsIndex] = {
-          offset:
-            (diffPosition -
-              auxdataDiffObjects[auxdataDiffIndex].diffStart -
-              2) /
-            2,
-          value: `0x${auxdataDiffObjects[auxdataDiffIndex].real}`,
+
+    // For each bytecodeDiffPosition, test every auxdataDiff that we haven't already mapped
+    for (let idx = 0; idx < auxdataDiffs.length; idx++) {
+      // The first CompiledContractCborAuxdata element key is '1'
+      const resultIndex = idx + 1;
+
+      // If we already mapped this auxdata, skip
+      if (result[resultIndex] !== undefined) continue;
+
+      const diff = auxdataDiffs[idx];
+      // If in position `pos` of the originalBytecode we find the `auxdataDiff.real` value
+      if (bytecodeIncludesAuxdataDiffAt(originalBytecode, diff, pos)) {
+        // Store the CompiledContractCborAuxdata element
+        result[resultIndex] = {
+          offset: (pos - diff.diffStart - 2) / 2,
+          value: `0x${diff.real}`,
         };
+        // One auxdata for each bytecodeDiffPosition. See https://github.com/ethereum/sourcify/issues/1980
```

Comment:
I think we mean first here

```suggestion
        // Match the first auxdata for each bytecodeDiffPosition. If there are multiple identical cborAuxdata, without the break it will always match the last one. With first, it will be "mapped" in the `result[resultIndex]` See https://github.com/ethereum/sourcify/issues/1980
```